.github/aggregate.js

```diff
@@ -0,0 +1,60 @@
+const fs = require('fs')
+const process = require('process')
+const path = require('path')
+
+// params: list of json reports to aggregate
+const inputs = process.argv.slice(2)
+
+const processReport = (filePath) => {
+  const resolvedPath = path.isAbsolute(filePath) ? filePath : path.resolve(process.cwd(), filePath);
+  const json = JSON.parse(fs.readFileSync(resolvedPath, 'utf8'))
+
+  // extract the TestMetadata
+  const { TestMetadata, ...rest } = json
+  const { time, meta } = TestMetadata || {}
+  const { version, job_url, gateway_url } = meta || {}
+
+  // and the data we care about.
+  // keep only the tests at the root
+  const entries = Object.entries(rest)
+  const roots = entries
+    .filter(([_, value]) => value.path.length === 1)
+    .map(([key, value]) => {
+      const result = {
+        outcome: value.outcome,
+      }
+
+      return [key, result]
+    })
+
+  return [
+    gateway_url,
+    {
+      metadata: {
+        time, version, job_url, gateway_url,
+      },
+      results:
+        Object.fromEntries(
+          roots,
+        ),
+    },
+  ]
+}
+
+const main = async () => {
+  const results = {}
+
+  inputs.forEach((filePath) => {
+    const [name, report] = processReport(filePath)
+    results[name] = report
+  })
+
+  fs.writeFileSync(1, JSON.stringify(results, null, 2))
```

Comment:
I'm not familiar with that pattern, do you mean creating a `Map` instance and inserting elements in order?

Reply:
Your deduction is incorrect it should be:

> since the **JSON Specification** provides no guarantees of keys being sorted. The JSON representation in report.json can have keys in random order.

In modern JS, the `Map` class orders keys by insertion, but unless you have insider knowledge about node's JSON serialization we can't solve this with JSON.

I would stick to regular JSON for now and revisit if / when we need that ordering (maybe when we want to render the table).